frontend/app/routes/protected/person-case/parent-details.tsx

```diff
@@ -0,0 +1,389 @@
+          />
+          <div className="mt-8 flex flex-row-reverse flex-wrap items-center justify-end gap-3">
+            <Button name="action" value="next" variant="primary" id="continue-button" disabled={isSubmitting}>
+              {t('protected:person-case.next')}
+            </Button>
+            <Button name="action" value="back" id="back-button" disabled={isSubmitting}>
+              {t('protected:person-case.previous')}
+            </Button>
+          </div>
+        </fetcher.Form>
+      </FetcherErrorSummary>
+    </>
+  );
+}
+
+type detailsType = {
+  unavailable?: boolean;
+  givenName?: string;
+  lastName?: string;
+  country?: string;
+  province?: string;
+  city?: string;
+};
+
+interface ParentInformationProps {
+  defaultFormValues?: detailsType[];
+  errors?: Record<string, [string, ...string[]] | undefined>;
+  amount: number;
+  setAmount: Dispatch<SetStateAction<number>>;
+}
+
+function ParentInformation({ defaultFormValues, errors, amount, setAmount }: ParentInformationProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const maxParents = 4;
+  const parentAmount = amount > 0 ? amount : 1;
+  const canAddParent = parentAmount < maxParents;
+
+  const onAddParent = () => {
+    if (!canAddParent) return;
+    setAmount(parentAmount + 1);
+  };
+
+  const onCloseParent = () => {
+    setAmount(parentAmount - 1);
+  };
+
+  return (
+    <>
+      <div className="space-y-10">
+        {Array.from({ length: parentAmount }).map((_, index) => (
+          <ParentForm
+            key={index}
+            index={index}
+            defaultValues={defaultFormValues?.[index]}
+            errors={errors}
+            onClose={index > 0 && index + 1 === parentAmount ? onCloseParent : undefined}
+          />
+        ))}
+      </div>
+      {canAddParent && (
+        <Button size="lg" type="button" variant="link" endIcon={faPlus} className="mt-3 px-3" onClick={onAddParent}>
+          {t('protected:parent-details.add-parent')}
+        </Button>
+      )}
+    </>
+  );
+}
+
+interface ParentFormProps {
+  index: number;
+  defaultValues?: detailsType;
+  errors?: Record<string, [string, ...string[]] | undefined>;
+  onClose?: () => void;
+}
+
+function ParentForm({ index, defaultValues, errors, onClose }: ParentFormProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const [unavailable, setUnavailable] = useState<boolean>(defaultValues?.unavailable ?? false);
+  const [country, setCountry] = useState<string | undefined>(defaultValues?.country);
+  const canadaCountryCode = 'CAN';
+
+  const countryOptions = (['select-option', ...COUNTRIES] as const).map((value) => ({
+    value: value === 'select-option' ? '' : value,
+    children: t(`protected:countries.${value}`),
+  }));
+
+  const provinceOptions = (['select-option', ...PROVINCES] as const).map((value) => ({
+    value: value === 'select-option' ? '' : value,
+    children: t(`protected:provinces.${value}`),
+  }));
```

Comment:
```suggestion
  const provinceOptions = (['select-option', ...PROVINCES] as const)
    .map((value) => ({
      value: value === 'select-option' ? '' : value,
      children: t(`protected:provinces.${value}`),
    }))
    .sort(({ children: left }, { children: right }) => left.localeCompare(right, lang));
```

Reply:
Sorry.. I added some comments in vscode using the github extension and it seems to have messed up a bunch of the line numbers.

Anyway, the idea here is to sort the `<select>` by province name, using the correct locale.

Reply:
I just realized that the code I suggested doesn't move the "select option" option to the top.

Reply:
``` ts
    .sort((left, right) => {
      if (left.value === '') return -1; // sort empty string to the top
      if (right.value === '') return 1; // sort empty string to the top
      return left.children.localeCompare(right.children, lang);
    });
```